File: samples/java/Bingo/Server/Matchmaking/src/main/java/systems/zlink/samples/bingo/server/matchmaking/BingoMatchmakerIdleTimerHandler.java
package systems.zlink.samples.bingo.server.matchmaking;

import systems.zlink.framework.spots.ZLinkTimerTick;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public final class BingoMatchmakerIdleTimerHandler {
    public CompletionStage<Void> handle(BingoMatchmaker spot, ZLinkTimerTick tick) {
        spot.closeIfIdle();
        return CompletableFuture.completedFuture(null);
    }
}

File: samples/kotlin/Bingo/Server/Matchmaking/src/main/kotlin/systems/zlink/samples/kotlin/bingo/server/matchmaking/BingoMatchmakerIdleTimerHandler.kt
package systems.zlink.samples.kotlin.bingo.server.matchmaking

import systems.zlink.framework.kotlin.ZLinkSuspendingSpotTimerHandler
import systems.zlink.framework.spots.ZLinkTimerTick

class BingoMatchmakerIdleTimerHandler : ZLinkSuspendingSpotTimerHandler<BingoMatchmaker> {
    override suspend fun handle(spot: BingoMatchmaker, tick: ZLinkTimerTick) {
        spot.closeIfIdle()
    }
}
